proxy/src/coco/git_helper.rs

```diff
@@ -0,0 +1,57 @@
+//! git-remote-rad git helper related functionality.
+
+use crate::config;
+
+use crate::error;
+use std::fs;
+use std::os::unix::fs::PermissionsExt;
+use std::path::Path;
+
+/// Trait for extending the Path module.
+trait IsExecutable {
+    /// Returns true if it is an executable.
+    fn is_executable(&self) -> bool;
+}
+
+/// Implements isExecutable for Path.
+impl IsExecutable for Path {
+    fn is_executable(&self) -> bool {
+        let metadata = match self.metadata() {
+            Ok(metadata) => metadata,
+            Err(_) => return false,
+        };
+        let permissions = metadata.permissions();
+        permissions.mode() & 0o111 != 0
+    }
+}
+
+/// Checks if the git-remote-rad helper is in a stable location and has the
+/// executable flag, if not copies the executable to the right place.
+///
+/// # Errors
+///
+///   * Could not get the path to directory where helper binaries should be stored.
+///   * Could not get the current working directory.
+///   * Could not create the path to binary directory.
+///   * Could not copy helper executable to the binary directory.
+pub fn setup() -> Result<(), error::Error> {
+    log::info!("Making sure git-remote-rad helper is set up");
+
+    let helper_binary_name = "git-remote-rad";
```

Comment:
This can be a `pub const` at the top of the file.

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/769/commits/7c249eaa129c1cfd5996b6b76fcee1ffaf6a2046